### ISSUE TITLE
Make `banker.SendCoins` take variadic `Coin` argument

### ISSUE THREAD

**Issue by leohhhn** [gnolang/gno#1676]
## Description

As the title says. Instead of having to build a slice each time you want to send coins, I propose we change the banker API to take in a variadic Coin argument.

Before:
`func (ba bankAdapter) SendCoins(from, to Address, amt Coins)`

After: 
`func (ba bankAdapter) SendCoins(from, to Address, amts ...Coin)`

[Here](https://github.com/gnolang/gno/blob/9ce5984635d9756c0b5f7d208b6ec30578cfcb08/gnovm/stdlibs/std/banker.gno#L56) is the code.

**Comment by thehowl** [comment]
Note, the `Coin` and `Coins` type are missing many useful methods and constructors from their original form in tm2/pkg/std: https://github.com/gnolang/gno/blob/master/tm2/pkg/std/coin.go

IMO a better approach is to add better constructors for the types; using the `Coins` type is useful as it has useful methods to do logic on coins

**Comment by leohhhn** [comment]
I was more referring to the fact that when you want to send a single coin with the banker, you always need to pack it inside the `Coins` struct, which can get a bit annoying. Example:

```
realmAddr := std.CurrentRealm().Addr()
coinToSend := banker.GetCoins(realmAddr).AmountOf("ugnot")
banker.SendCoins(realmAddr, to, std.Coins{{ "ugnot", coinToSend }})

----

// bit nicer if:
banker.SendCoins(realmAddr, to, std.Coin{"ugnot", coinToSend})

// and can do
coins := std.Coins{...}
banker.SendCoins(realmAddr, to, coins...)
```

BTW, I am working on a small PR to introduce some more functionality to `Coins/Coin` in Gno by porting some stuff from `coin.go`.

**Comment by thehowl** [comment]
Yeah, I get you, what I'm saying is that `Coins` is a useful type and maybe we should tackle the underlying issue here by making its constructors useful.

**Comment by leohhhn** [comment]
Solution for this was to implement a variadic constructor for the Coins type. Implemented in #2104
